packages/jest-jasmine2/src/jasmineAsyncInstall.ts

```diff
@@ -104,12 +109,18 @@ function promisifyIt(
       const spec = originalFn.call(env, specName);
       spec.pend('not implemented');
       return spec;
+    } else if (typeof fn !== 'function') {
+      // Pass non-functions to Jest, which throws a nice error.
+      return originalFn.call(env, specName, fn, timeout);
     }
 
     const hasDoneCallback = fn.length > 0;
 
     if (hasDoneCallback) {
-      return originalFn.call(env, specName, fn, timeout);
+      // Give the function a name so it can be detected in call stacks, but
+      // otherwise Jasmine will handle it.
+      const asyncJestTestWithCallback = (done: DoneFn) => fn(done);
+      return originalFn.call(env, specName, asyncJestTestWithCallback, timeout);
```

Comment:
Technically this changes effect slightly. If you passed in a non-function object with a `length` property before (e.g. a non-empty string, as in [this test](https://github.com/facebook/jest/blob/a2f17006059b6ef39e5e4f09b4b40f241a680fd8/packages/jest-jasmine2/src/__tests__/itTestError.test.ts#L23-L30)), we’d either do what’s happening here if it had a length, or wind up wrapping it with `asyncJestTest`, which would then cause an error later when the test ran and it tried to call the non-function.

I *think* this is the right change to make, though, since it seems like it’s addressing those possible other error cases. (There’s an edge-case here: you could have passed an object with a `call()` method that I think would have worked before, but would not with this change. I’m not sure if that’s really supposed to be supported, though.)